extensions/wikia/SitemapPage/tests/SitemapPageTest.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+class SitemapPageTest extends WikiaBaseTest {
+
+	const TEST_CITY_ID = 79860;
+	const TEST_ARTICLE = 'test 123';
+	const TEST_ARTICLE_SITEMAP = 'sitemap';
+
+	public function setUp() {
+		$this->setupFile = dirname( __FILE__ ).'/../SitemapPage.setup.php';
+		parent::setUp();
+	}
+
+	/**
+	 * @dataProvider getLimitPerPageDataProvider
+	 */
+	public function testGetLimitPerPage( $totalWikis, $exp ) {
+		$mock = $this->getMock( 'SitemapPageModel', [ 'getTotalWikis' ] );
+		$mock->expects( $this->any() )
+			->method( 'getTotalWikis' )
+			->will( $this->returnValue( $totalWikis ) );
+		$limit = $mock->getLimitPerPage();
+		$this->assertEquals( $exp, $limit );
+	}
+
+	public function getLimitPerPageDataProvider() {
+		return [
+			[ 0, 100 ],
+			[ '100', 100 ],
+			[ 999999, 100 ],
+			[ '1000000', 100 ],
+			[ '1000001', 150 ]
+		];
+	}
+
+	/**
+	 * @dataProvider getLimitPerListDataProvider
+	 */
+	public function testGetLimitPerList( $level, $exp ) {
+		$limitPerPage = 100;
+		$mock = $this->getMock( 'SitemapPageModel', [ 'getLimitPerPage' ] );
+		$mock->expects( $this->any() )
+			->method( 'getLimitPerPage' )
+			->will( $this->returnValue( $limitPerPage ) );
+		$limit = $mock->getLimitPerList( $level );
+		$this->assertEquals( $exp, $limit );
+	}
+
+	public function getLimitPerListDataProvider() {
+		return [
+			[ 1, 10000],
+			[ 2, 100]
```

Comment:
Any tests for level 0 or a negative level? Level 3, 4, 5, etc? Edge/corner cases?